cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -160,18 +160,13 @@ func TestEtcdSupportedVersion(t *testing.T) {
 			expectedError:     errors.New("Unsupported or unknown Kubernetes version(1.99.0)"),
 		},
 		{
-			kubernetesVersion: "1.12.1",
```

Comment:
ideally we shouldn't remove the test for the version in the support skew.
posted another comment bellow.

Reply:
how about we add a new constant:
`MinimumKubernetesSkewVersion = ...1.12.0`

cc @rosti @fabriziopandini

Reply:
+1 it's can help on upgrade tests

Reply:
+1 for keeping the test. I am thinking, however, what is less bad - having another constant or having a trivial piece of code, that calculates it from `constants.MinimumControlPlaneVersion`?

Reply:
here is what i propose:
- add this new constant (which is technically a variable) in constants.go
(AFAIK it will be used in more than one place)
- in another PR, based on CurrentKubernetesVersion calculate the rest of the version variables.
(i wonder if we can use `init()` in constants.go here?)

Reply:
@neolit123 I like your idea very much. We don't need to use `init()` though. Global var initializations are executed before the `init()` calls.

Reply:
ok, @yagonobre could you please add `MinimumKubernetesSkewVersion = ..."1.12"` to constants.go as part of this PR and keep the 3rd test in TestEtcdSupportedVersion.

in a separate PR(s) we can simplify:
1) upgrade tests
2) optionally make versions variables be based on CurrentKubernetesVersion

but with 2) i do realize there is one problem. in case we bump k8s to v2 this naive logic will not work:
`MinimumKubeletVersion = CurrentKubernetesVersion.WithMinor(CurrentKubernetesVersion.Minor() - 1)`
and it needs a utility function.

Reply:
Yes, we should be v2 safe (though we aren't in all other places ATM - `EtcdSupportedVersion` is one such place).